arangod/IResearch/IResearchLink.cpp

```diff
@@ -2010,7 +2226,8 @@ Result IResearchLink::remove(transaction::Methods& trx,
     ctx = ptr.get();
     state.cookie(key, std::move(ptr));
 
-    if (!ctx || !trx.addStatusChangeCallback(&_trxCallback)) {
+    if (!ctx || !trx.addStatusChangeCallback(&_trxCallback) ||
+        !trx.addPreCommitCallback(&_trxPreCommit)) {
```

Comment:
if add precommit failed, should we remove status change or not?

Reply:
I guess not. As transaction will be aborted with error anyway.